src/common/git/with-git-hooks.ts

```diff
@@ -9,7 +14,26 @@ export interface WithGitHooks {
   /**
    * Include a default git hook that checks commit message.
    *
+   * @default [HuskyRules.COMMIT_MESSAGE]
+   */
+  readonly huskyRules?: Array<HuskyRule>
```

Comment:
@quesabe do you think we can change the interface to this?
```typescript
haskyRules: {
  checkCargo?: CargoOptions
  commitMsg?: CommitMsgOptions
}
```

We preserve kind of enum for the rules at the level of permitted object properties, while also keep the options perfectly typed.

Reply:
I think it would work. I just did it the way it's done with dependabot within projen GitHub component. Your proposal seems cleaner, so let's use this setup.